src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -707,6 +706,121 @@ void ep_rt_aot_os_environment_get_utf16 (dn_vector_ptr_t *env_array)
 #endif
 }
 
+// Generate cryptographically strong random bytes.
+// Return 0 on success, -1 on failure.
+// copying code from SystemNative_GetCryptographicallySecureRandomBytes with slight modification
+int32_t aot_get_cryptographically_secure_random_bytes(uint8_t* buffer, int32_t bufferLength)
+{
+    assert(buffer != NULL);
+#ifdef __linux__
+    static volatile int rand_des = -1;
+    static bool sMissingDevURandom;
+
+    if (!sMissingDevURandom)
+    {
+        if (rand_des == -1)
+        {
+            int fd;
+
+            do
+            {
+                fd = open("/dev/urandom", O_RDONLY);
```

Comment:
Can we use [`getrandom`](https://man7.org/linux/man-pages/man2/getrandom.2.html) instead?

Reply:
`getrandom` requires glibc 2.25. Our minimum glibc for .NET 8 is glibc 2.23 (https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md#libc-compatibility).

So we would need to have all the code here anyway to provide fallback for the older glibc. It is non-trivial to get it setup. https://github.com/dotnet/runtime/pull/1433 tried to do that, but it was never finished.